hummingbot/core/gateway/gateway_http_client.py

```diff
@@ -467,38 +468,56 @@ async def cancel_evm_transaction(
             "nonce": nonce
         })
 
-    async def amm_trade(
-            self,
-            chain: str,
-            network: str,
-            connector: str,
-            address: str,
-            base_asset: str,
-            quote_asset: str,
-            side: TradeType,
-            amount: Decimal,
-            price: Decimal,
-            nonce: Optional[int] = None,
-            max_fee_per_gas: Optional[int] = None,
-            max_priority_fee_per_gas: Optional[int] = None
-    ) -> Dict[str, Any]:
-        # XXX(martin_kou): The amount is always output with 18 decimal places.
-        request_payload: Dict[str, Any] = {
-            "chain": chain,
-            "network": network,
-            "connector": connector,
-            "address": address,
-            "base": base_asset,
-            "quote": quote_asset,
-            "side": side.name,
-            "amount": f"{amount:.18f}",
-            "limitPrice": str(price),
-            "allowedSlippage": "0/1",  # hummingbot applies slippage itself
-        }
-        if nonce is not None:
-            request_payload["nonce"] = int(nonce)
-        if max_fee_per_gas is not None:
-            request_payload["maxFeePerGas"] = str(max_fee_per_gas)
+async def amm_trade(
+    self,
+    chain: str,
+    network: str,
+    connector: str,
+    address: str,
+    base_asset: str,
+    quote_asset: str,
+    side: TradeType,
+    amount: Decimal,
+    price: Decimal,
+    allowedSlippage: str = "0/1",  # Default value is "0/1"
+    nonce: Optional[int] = None,
+    max_fee_per_gas: Optional[int] = None,
+    max_priority_fee_per_gas: Optional[int] = None
+) -> Dict[str, Any]:
+    # Check if the amount and price are valid numbers
+    if not isinstance(amount, (int, float, Decimal)) or not isinstance(price, (int, float, Decimal)):
+        raise ValueError("Amount and price must be numbers")
+
+    # Check if the amount and price are not null
+    if amount is None or price is None:
+        raise ValueError("Amount and price cannot be null")
+
+    # Check if the price is too small
+    if price < 1e-7:
+        raise ValueError("Price is too small")
+
+    # Format the amount and price
+    getcontext().prec = 28  # Set the precision high enough
```

Comment:
This is not needed, default precision in python is already up to 28 significant figures. Also please be careful when changing global scope specs since it could affect other logics. Please use `localcontext()` with `with` statement so the change stay within scope. More info: https://docs.python.org/3/library/decimal.html#decimal.localcontext

Reply:
ok

Reply:
is my rest of code is ok